### DISCUSSION TITLE
Static Caching invalidation not working

### DISCUSSION THREAD

**Discussion by RyanRoberts** [statamic/cms#10413] [Troubleshooting] [answered]
I'm not sure if this is a bug or something I'm doing. 

I have a site using the "half" caching strategy and I noticed content changes in globals weren't being updated on the site, then I noticed "blog" post titles weren't being updated elsewhere too. 

Going by [this documentation](https://statamic.dev/static-caching#invalidation) I added the following to my static_caching.php file:

```php
'invalidation' => [

	'class' => null,

	'rules' => [
		'globals' => [
			'seo' => [
				'urls' => [
					'/*'
				]
			]
		],
		'collections' => [
			'articles' => [
				'urls' => [
					'/*'
				]
			]
		],
		'navigation' => [
			'primary' => [
				'urls' => [
					'/*'
				]
			]
		]
	],

],
```

However, nothing changed. The cache still wasn't being invalidated when one of those changed.

On the other hand this works:

```php
'invalidation' => [

	'class' => null,

	'rules' => 'all',

],
```

I have no idea where to go from here 💩

**Comment by robdekort** [comment]
Your config looks good. Were these global changes coming from your `seo` global or another one? You have to specify them all.

**Reply by RyanRoberts** [reply]
That’s right. All three invalidation rules there relate to items that should be invalidating the cache.

Am I right I think it’s the global “set” you add a rule for and anything in that set should trigger the invalidation?

**Reply by robdekort** [reply]
> That’s right. All three invalidation rules there relate to items that should be invalidating the cache.
> 
> Am I right I think it’s the global “set” you add a rule for and anything in that set should trigger the invalidation?

Exactly, it should invalidate when the SEO set gets saved. It will ignore any other sets. So I'm not sure why you're running into this.

**Reply by RyanRoberts** [reply]
I'll spin up a clean Statamic 5 environment and see if I can replicate it with minimal customisation.

**Reply by RyanRoberts** [reply]
@robdekort I've managed to narrow it down. Doesn't look like a bug but something to do with my server config. It works fine locally (caching enabled) but on staging and production (both managed via Ploi.io) specifying the invalidation other than all just doesn't work. 

Curious 🤔

**Reply by ryanmitchell** [reply]
Do you have a queue specified (other than sync) and if so do you have a queue worker running in ploi? Invalidation will run on the queue if one is specified.

**Reply by robdekort** [reply]
Interesting. I use Ploi as well but haven’t run into this. If you ever find out I’d love to know.

**Reply by robdekort** [reply]
Ah yeah. Good call @ryanmitchell

**Reply by RyanRoberts** [reply]
@ryanmitchell I don't think I do, not one set in Ploi itself. Should I specify one?

**Reply by ryanmitchell** [reply]
It'll be in your .env file under `QUEUE_CONNECTION`

**Reply by RyanRoberts** [reply]
OK that's set it `QUEUE_CONNECTION=sync`. 

Sorry I'm very much a Statamic newbie.

**Reply by ryanmitchell** [reply, marked as answer]
Thats not it then - I didnt notice you mentioning `all` worked initially so im probably sending you down the wrong path.
Does your APP_URL in the .env match your staging URL?

**Reply by RyanRoberts** [reply]
That's the one! 🎉 

The url was http instead of https, I guess it was generated before I'd applied the SSL certificate and never thought to look. Thanks @ryanmitchell @robdekort for the help!

**Reply by robdekort** [reply]
Woohoo
